File: Backend/src/test/java/lan/dk/podcastserver/utils/form/MovingItemInQueueFormTest.java
package lan.dk.podcastserver.utils.form;

import org.junit.Test;

import java.util.UUID;

import static lan.dk.podcastserver.assertion.Assertions.assertThat;

/**
 * Created by kevin on 15/07/15 for Podcast Server
 */
public class MovingItemInQueueFormTest {

    @Test
    public void should_generate_object() {
        /* Given */
        MovingItemInQueueForm movingItemInQueueForm = new MovingItemInQueueForm();
        UUID id = UUID.randomUUID();

        /* When */
        movingItemInQueueForm.setId(id);
        movingItemInQueueForm.setPosition(2);

        /* Then */
        assertThat(movingItemInQueueForm)
                .hasId(id)
                .hasPosition(2);
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/utils/form/MovingItemInQueueFormTest.kt
package com.github.davinkevin.podcastserver.utils.form

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.util.*
import java.util.function.Function

internal class MovingItemInQueueFormTest {

    val id = Function<MovingItemInQueueForm, Any> { it.id }
    val position = Function<MovingItemInQueueForm, Any> { it.position }

    @Test
    fun `should instantiate and define fields`() {
        /* Given */
        val uuid = UUID.randomUUID()
        /* When */
        val form = MovingItemInQueueForm(id = uuid, position = 5)
        /* Then */
        assertThat(form)
                .extracting(id, position)
                .containsExactly(uuid, 5)
    }

    @Test
    fun `should be equal and has same hashcode`() {
        /* Given */
        val id = UUID.randomUUID()
        /* When */
        val form1 = MovingItemInQueueForm(id = id, position = 5)
        val form2 = MovingItemInQueueForm(id = id, position = 5)
        /* Then */
        assertThat(form1).isEqualTo(form2)
        assertThat(setOf(form1, form2)).hasSize(1)
    }
}

